apps/contributions/filters.py

```diff
@@ -60,13 +61,21 @@ def filter_queryset_by_rp(self, queryset, revenue_program_id: str):
             | Q(contribution_metadata__contains={"revenue_program_id": revenue_program_id})
         )
 
+    def filter_intervals(self, queryset, request):
+        if (interval := request.query_params.get("interval")) == ContributionInterval.ONE_TIME.value:
+            return queryset.filter(interval=ContributionInterval.ONE_TIME)
+        elif interval == "recurring":
+            return queryset.exclude(interval=ContributionInterval.ONE_TIME)
+        return queryset
+
```

Comment:
Minor suggestion, but this is a good place for match/case:

```suggestion
        match request.query_params.get("interval"):
            case ContributionInterval.ONE_TIME.value:
                return queyrset.filter(interval=ContributionInterval.ONE_TIME)
            case "recurring":
                return queryset.exclude(ContributionInterval.ONE_TIME)
            case _:
                return queryset
```